### ISSUE TITLE
Issue calculating medians in gegen pctile and gcollapse w/ small groups

### ISSUE THREAD

**Issue by mdroste** [mcaceresb/stata-gtools#38]
Medians appear to be calculated wrong in at least this simple case:

```
clear all
set obs 3
gen x = 1
replace x = 3 in 2
replace x = 5 in 3
egen med_egen = pctile(x), p(50)
gegen med_gegen = pctile(x), p(50)
list
```

The same problem also emerges in gcollapse:

```
clear all
set obs 3
gen x = 1
replace x = 3 in 2
replace x = 5 in 3
tempfile t1
save `t1'
collapse (p50) x
list
use `t1', clear
gcollapse (p50) x
list
```

**Comment by mcaceresb** [comment]
I think I introduced the bug [here](https://github.com/mcaceresb/stata-gtools/issues/37). The weighted version works fine.

**Comment by mcaceresb** [comment]
Scratch that. I introduced this when trying to deal with numerical precision issues a number of releases ago. I made a special case for when the quantile requested coincided with the largest value; however, I did not write the exception correctly. I think every time you request a quantile between 100 - 150 / N and 100 - 100 / N you get this error. For N = 3, you get the wrong answer for p50 through p66.7.
